File: 2026/07-July/06-Monday/solution.java
import java.util.*;

class Solution {
    public int removeCoveredIntervals(int[][] intervals) {

        Arrays.sort(intervals, (a, b) -> {
            if (a[0] == b[0]) {
                return Integer.compare(b[1], a[1]);
            }
            return Integer.compare(a[0], b[0]);
        });

        int answer = 0;
        int farthest = -1;

        for (int[] current : intervals) {
            if (current[1] > farthest) {
                answer++;
                farthest = current[1];
            }
        }

        return answer;
    }
}

File: 2026/07-July/06-Monday/solution.cs
using System;

public class Solution
{
    public int RemoveCoveredIntervals(int[][] intervals)
    {
        Array.Sort(intervals, (a, b) =>
        {
            if (a[0] == b[0])
                return b[1].CompareTo(a[1]);

            return a[0].CompareTo(b[0]);
        });

        int answer = 0;
        int farthestEnd = -1;

        foreach (var interval in intervals)
        {
            if (interval[1] > farthestEnd)
            {
                answer++;
                farthestEnd = interval[1];
            }
        }

        return answer;
    }
}
